### DISCUSSION TITLE
Is there an interface to display an error in the panel?

### DISCUSSION THREAD

**Discussion by tillsteinbach** [gapitio/gapit-htmlgraphics-panel#91] [Q&A]
I can `throw new Error()` but it will display something like "error in onRender". Would be great if there is a real interface to show errors to the user.

**Comment by ZuperZee** [comment]
Currently the error interface is mostly meant for the developer to make it easier to code. I'll open an enhancement issue for this ([issue 92](https://github.com/gapitio/gapit-htmlgraphics-panel/issues/92)), but it will probably take a long time before I start on implementing this (maybe someone else opens a PR for it :D).

One of the reasons why it would be hard to add a custom user error interface (calling it that :D) is the interface might block parts of the panel where the data is (the current error interface does that, but users aren't actually supposed to see it, just developers).

Just adding a way I usually display "errors".
Instead of throwing an error I usually display a message for where the data is supposed to be. This doesn't work for everything (like screenshots and such, but here is probably where you want the custom user error interface) and you will have to make a custom error interface in the code for those.
![image](https://user-images.githubusercontent.com/39485579/155478760-63006e8f-a787-426e-83e9-0da666ae7ded.png)

Just some screenshots of the current error interface meant for the developers.
![image](https://user-images.githubusercontent.com/39485579/155475858-64dfceb7-1864-4c5b-9e65-bf384482ebdb.png)
![image](https://user-images.githubusercontent.com/39485579/155476517-d28f8539-ae60-405a-8d2d-5bb025432fa5.png)

**Reply by tillsteinbach** [reply]
yes, I used the second approach which is working fine, would only be cool to be able to change the title. Maybe just by having a custom Exception type with title.

**Reply by ZuperZee** [reply]
I could add a custom exception, but I don't know how to handle events with exceptions. Since my code can't listen to exceptions in events like onclick, onchange, ETC.
Since you're already throwing a custom exception, a setError function might work better. I'll take a look at how it can be implemented.
